nifi-extension-bundles/nifi-poi-bundle/nifi-poi-services/src/main/java/org/apache/nifi/excel/RowIterator.java

```diff
@@ -129,4 +151,37 @@ private boolean hasExhaustedRows() {
         }
         return exhausted;
     }
+
+    private boolean isXSSFExcelFile(InputStream in, final String password) {
+        final FileMagic fileType;
+        try {
+            fileType = FileMagic.valueOf(in);
+        } catch (final IOException e) {
+            throw new ProcessException("Failed to determine file type", e);
+        }
+
+        if (fileType == FileMagic.OOXML) {
+            // Unencrypted XLSX file
+            return true;
+        }
+        if (fileType != FileMagic.OLE2) {
+            // Not an Excel file
+            return false;
+        }
+        if (password == null) {
+            // If there is no password on the OLE2 file, then it's an XLS file. Otherwise, the type must be further
+            // investigated.
+            return false;
+        }
+
+        DirectoryNode root;
+        try {
+            root = new POIFSFileSystem(new NonCloseableInputStream(in)).getRoot();
```

Comment:
In addition I think the creation of `POIFSFileSystem` and the assignment to `root` should be done on separate lines. Also the return statement should be all done in the try block as if the instantiation of `POIFSFileSystem` throws an IOException the return statement is never reached.

```suggestion
        DirectoryNode root;
        try {
           final POIFSFileSystem poifsFileSystem = new POIFSFileSystem(new NonCloseableInputStream(in)).getRoot();
           final DirectoryNode root = poifsFileSystem.getRoot() ;
           return root.hasEntryCaseInsensitive(DEFAULT_POIFS_ENTRY) || root.hasEntryCaseInsensitive(OOXML_PACKAGE);
           }
```